### PR TITLE
docs: explain slashing mechanism

### LINKED ISSUE #11: Explain Slashing in the Docs
Feedback from one of the validators is to explain the slashing mechanics in consensus using the docs.

### PR DESCRIPTION
**Pull request by Mackenzie-OO7**
Resolves #11

- [x] New and updated code has appropriate documentation
- [x] New and updated code has new and/or updated testing
- [x] Required CI checks are passing
- [x] Visual proof for any user facing features like CLI or documentation updates
- [x] Linked issues closed with keywords

## Summary by CodeRabbit

- **New Features**
  - Introduced "Slashing Mechanics" documentation for validators in the sidebar.
- **Documentation**
  - Added detailed information on slashing conditions and impacts for Celestia validators and their delegators.

### CODE REVIEW

File: developers/slashing.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+1. **Downtime**: If a validator is offline for more than 95% of a rolling window
+   of the last 10,000 blocks, they will lose 0.01% of their stake.
+   During this slashing event, the validator is removed from the validator set
+   temporarily and spends time in "jail" for a period of 10 mins during which
+   they will be unable to propose new blocks or earn rewards.
+   After the jail period, the validator can send an unjail request to
+   rejoin the validator set.
+
+2. **Double Signing**: This is a more severe offense and results in a larger
```
Comment (jcstein) [inline developers/slashing.md:26]:
```suggestion
2. **Double signing**: This is a more severe offense and results in a larger
```

File: developers/slashing.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+   percentage loss. If a validator engages in double signing, the validator
+   will lose 5% of their stake and their stake will be returned to them.
+   The validator will be permanently removed from the validator set and
+   will not have the ability to unjail. Delegators bonded to that validator will
+   automatically enter the unbounding period and can delegate to another
```
Comment (jcstein) [inline developers/slashing.md:31]:
```suggestion
   automatically enter the unbonding period and can delegate to another
```

File: developers/slashing.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+   validator after they have been unbounded.
```
Comment (jcstein) [inline developers/slashing.md:32]:
```suggestion
   validator after they have been unbonded.
```

File: developers/slashing.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+of a validators stake each time they act harmfully towards the network.
+
+Celestia is built with the Cosmos-SDK and implements the module `x/slashing`.
+
+If a Validator gets slashed, delegators bonded to that validator will also
+have the same percentage of their delegated funds slashed.
+
+The following are the conditions for a validator to get slashed:
+
+1. **Downtime**: If a validator is offline for more than 95% of a rolling window
```
Comment (jcstein) [inline developers/slashing.md:18]: https://celestiaorg.github.io/celestia-app/specs/params.html could you set the params based on these and link accordingly?

**Review by jcstein — CHANGES_REQUESTED** [review]

File: developers/slashing.md (hunk 4 above)
Comment (Mackenzie-OO7) [reply developers/slashing.md:18]: Sure! Thanks for reviewing

Comment (jcstein) [reply developers/slashing.md:18]: also, use values from the page linked above

Comment (Mackenzie-OO7) [reply developers/slashing.md]: Please what did you mean by "link accordingly"? There's no way to link each one to the parameter listed in the specs

Comment (jcstein) [reply developers/slashing.md]: I meant to link to individual params, but I realize that this isn't currently possible. Could you just link to the specs once in the page? thanks!

File: developers/slashing.md [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+2. **Double signing**: This is a more severe offense and results in a larger
+   percentage loss. If a validator engages in double signing, the validator
+   will lose 2% of their stake and their stake will be returned to them.
+   The validator will be permanently removed from the validator set and
+   will not have the ability to unjail. Delegators bonded to that validator will
+   automatically enter the unbonding period for 21 days, and can delegate to another
+   validator after they have been unbonded.
+
+For more details on the slashing parameters, refer to the
+[Celestia App Specifications](https://celestiaorg.github.io/celestia-app/specs/params.html#module-parameters)
```
Comment (jcstein) [inline developers/slashing.md:35]:
```suggestion
[Celestia App governance specifications](https://celestiaorg.github.io/celestia-app/specs/params.html#module-parameters)
```

**Comment by jcstein** [comment]
hi, apologies for the delay here! i need to review this again, the deploy pr preview will not be able to be ran as you are from outside the organization, though

File: developers/slashing.md [hunk 6]
```diff
@@ -0,0 +1,36 @@
+---
+description: This section covers the slashing mechanics for validators in Celestia.
+---
+
+# Slashing on Celestia
+
+Slashing is mechanism employed in PoS blockchains that is used to deter
```
Comment (rootulp) [inline developers/slashing.md:7]:
```suggestion
Slashing is a mechanism employed in Proof of Stake (PoS) blockchains that is used to deter
```

File: developers/slashing.md [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+and punish malicious behavior. It functions by removing a percentage
+of a validators stake each time they act harmfully towards the network.
+
+Celestia is built with the Cosmos-SDK and implements the module `x/slashing`.
+
+If a Validator gets slashed, delegators bonded to that validator will also
```
Comment (rootulp) [inline developers/slashing.md:13]:
```suggestion
If a validator gets slashed, delegators bonded to that validator will also
```

File: developers/slashing.md [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+   of the last 5,000 blocks, they will lose 0% of their stake.
+   During this slashing event, the validator is removed from the validator set
+   temporarily and spends time in "jail" for a period of 1 minute during which
+   they will be unable to propose new blocks or earn rewards.
+   After the jail period, the validator can send an unjail request to
+   rejoin the validator set.
+
+2. **Double signing**: This is a more severe offense and results in a larger
+   percentage loss. If a validator engages in double signing, the validator
+   will lose 2% of their stake and their stake will be returned to them.
```
Comment (rootulp) [inline developers/slashing.md:28]: [question] what does this mean? If the validator loses 2% of their stake and it's returned to them, then they aren't losing it.

Perhaps this means the remainder of their stake will be returned to them?

File: developers/slashing.md [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+
+Celestia is built with the Cosmos-SDK and implements the module `x/slashing`.
+
+If a Validator gets slashed, delegators bonded to that validator will also
+have the same percentage of their delegated funds slashed.
+
+The following are the conditions for a validator to get slashed:
+
+1. **Downtime**: If a validator is offline for more than 75% of a rolling window
+   of the last 5,000 blocks, they will lose 0% of their stake.
```
Comment (rootulp) [inline developers/slashing.md:19]: [blocking] the way this is phrased is incorrect. If a validator doesn't sign 75% of the last 5,000 blocks they will be jailed. So if they are offline for more than 25% of that rolling window, they will be jailed.

File: developers/slashing.md [hunk 10]
```diff
@@ -0,0 +1,36 @@
+---
+description: This section covers the slashing mechanics for validators in Celestia.
+---
+
+# Slashing on Celestia
+
+Slashing is mechanism employed in PoS blockchains that is used to deter
+and punish malicious behavior. It functions by removing a percentage
+of a validators stake each time they act harmfully towards the network.
```
Comment (rootulp) [inline developers/slashing.md:9]:
```suggestion
of a validator's stake each time they act harmfully towards the network.
```

File: developers/slashing.md [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+
+Celestia is built with the Cosmos-SDK and implements the module `x/slashing`.
```
Comment (rootulp) [inline developers/slashing.md:11]:
```suggestion
Celestia is built with the Cosmos SDK and uses the `x/slashing` module.
```

File: developers/slashing.md [hunk 12]
```diff
@@ -0,0 +1,36 @@
...
+2. **Double signing**: This is a more severe offense and results in a larger
+   percentage loss. If a validator engages in double signing, the validator
+   will lose 2% of their stake and their stake will be returned to them.
+   The validator will be permanently removed from the validator set and
+   will not have the ability to unjail. Delegators bonded to that validator will
+   automatically enter the unbonding period for 21 days, and can delegate to another
+   validator after they have been unbonded.
+
+For more details on the slashing parameters, refer to the
+[Celestia App governance specifications](https://celestiaorg.github.io/celestia-app/specs/params.html#module-parameters)
```
Comment (rootulp) [inline developers/slashing.md:35]: [optional] to drop "governance" here

```suggestion
[celestia-app specifications](https://celestiaorg.github.io/celestia-app/specs/params.html#module-parameters)
```

**Review by rootulp — COMMENTED** [review]
Thanks for the contribution! A few comments on the language used

File: .vitepress/config.ts [hunk 13]
```diff
@@ -728,6 +728,10 @@ function sidebarHome() {
                 },
               ],
             },
+            {
```
Comment (jcstein) [inline .vitepress/config.ts:731]: @Mackenzie-OO7 what do you think of putting this in the node operators section in celestia-app category?
screenshot 2024-01-24 at 17 06 55

Comment (Mackenzie-OO7) [reply .vitepress/config.ts:731]: hii, sorry for the delay. of course! not sure why I put it in the developers section in first place😅

Comment (jcstein) [reply .vitepress/config.ts:731]: no worries! i was about to move it myself and approve PR 😄

File: nodes/celestia-app-slashing.md [hunk 14]
```diff
@@ -0,0 +1,37 @@
+---
+description: This section covers the slashing mechanics for validators in Celestia.
+---
+
+# Slashing on Celestia
+
+Slashing is a mechanism employed in Proof of Stake (PoS) blockchains
```
Comment (jcstein) [inline nodes/celestia-app-slashing.md:7]:
```suggestion
Slashing is a mechanism employed in proof of stake (PoS) blockchains
```

**Review by jcstein — APPROVED** [review]

**Pull request merged**